### PR TITLE
Uses URL from ServerIdentity struct (JS only) issue #1791

### PR DESCRIPTION
**Pull request by nkcr**
Solves issue #1791. 
If an `Url` param is found in the toml file, `getWebSocketAddress()` will use it instead of simply transforming the current server address with a port increment of one.

This PR also adds a test case to check if a docker service is available and changes a strange condition in `spec/support/conondes.ts` that was failing the tests. It also documents how to use `js/cothority` as a local module, which was used in order to check the service against `dedis/student_18_explorer`.

### CODE REVIEW

File: external/js/cothority/README.md [hunk 1]
```diff
@@ -91,3 +93,45 @@ will be provided thanks to the [buffer package](https://www.npmjs.com/package/bu
...
+
+**Note**: Using the official command `yarn link` doesn't seem to work. We leave
+the steps at the end for informational purpose, but we found out that using symlink
+is simpler, clearer, and actually works.
+
+Build and Link the local `@dedis/cothority` module to a local app:
+
+```bash
+(cothority/external/js/cothority) $ npm run build
+$ ln -s /<cothority root path>/cothority/external/js/cothority/dist /<local app root path>/node_modules/@dedis/cothority
```
Comment (Gilthoniel) [inline external/js/cothority/README.md]: It seems easier to me to use `npm link @dedis/cothority` inside the local app root

File: external/js/cothority/README.md [hunk 2]
```diff
@@ -91,3 +93,45 @@ will be provided thanks to the [buffer package](https://www.npmjs.com/package/bu
 As this is a [polyfill](https://remysharp.com/2010/10/08/what-is-a-polyfill), please
 check that what you need is implemented or you will need to use a different approach. Of
 course for NodeJS, you will always get a [Buffer](https://nodejs.org/api/buffer.html).
+
+### Use a developpment version from an external app
```
Comment (Gilthoniel) [inline external/js/cothority/README.md]: `##` as it is not related to the Protobuf generation chapter

File: external/js/cothority/README.md [hunk 3]
```diff
@@ -91,3 +93,45 @@ will be provided thanks to the [buffer package](https://www.npmjs.com/package/bu
...
+3) From the root folder of the external app, link the new package
+
+```bash
+(external_app) $ yarn link "@dedis/cothority"
+```
+
+4) If needed, install the new linked pachage
+
+```bash
+(external_app) $ npm install --save @dedis/cothority
```
Comment (Gilthoniel) [inline external/js/cothority/README.md]: why using npm in that case ? I guess you mean `yarn add --save` ?

Comment (Gilthoniel) [reply external/js/cothority/README.md]: Also I'm not sure we should encourage using yarn as cothority and kyber are using npm. Or we should switch to yarn everywhere but it seems confusing and frustrating to mix the usage of both.

Comment (nkcr) [reply external/js/cothority/README.md]: I totally agree! Yarn is used in the student_18_explorer project while cothority uses npm. Thats something we should agree on some guidelines.

File: external/js/cothority/README.md (hunk 1 above)
Comment (nkcr) [reply external/js/cothority/README.md]: That doesn't work in my case. It links the `cothority` folder instead of the `cothority/dist` one.

Comment (Gilthoniel) [reply external/js/cothority/README.md]: I mean you run `npm run link` in the cothority library (the script will link the dist folder) and you navigate to the external project and use `npm link @dedis/cothority`.

File: external/js/cothority/README.md [hunk 4]
```diff
@@ -91,3 +93,32 @@ will be provided thanks to the [buffer package](https://www.npmjs.com/package/bu
 As this is a [polyfill](https://remysharp.com/2010/10/08/what-is-a-polyfill), please
 check that what you need is implemented or you will need to use a different approach. Of
 course for NodeJS, you will always get a [Buffer](https://nodejs.org/api/buffer.html).
+
+## Use a developpment version from an external app
```
Comment (jeffallen) [inline external/js/cothority/README.md]:
```suggestion
## Use a development version from an external app
```

File: external/js/cothority/src/network/proto.ts [hunk 5]
```diff
@@ -213,11 +214,16 @@ export class ServerIdentity extends Message<ServerIdentity> {
     }
 
     /**
-     * Convert the address of the server to match the websocket format
+     * Returns this.url if wset, otherwise converts the server
```
Comment (jeffallen) [inline external/js/cothority/src/network/proto.ts]:
```suggestion
     * Returns this.url if set, otherwise converts the server
```

**Pull request merged**